libs/parallelism/algorithms/include/hpx/parallel/algorithms/sort.hpp

```diff
@@ -376,14 +541,87 @@ namespace hpx { namespace parallel { inline namespace v1 {
             >::value
         )>
     // clang-format on
-    typename util::detail::algorithm_result<ExPolicy, RandomIt>::type sort(
-        ExPolicy&& policy, RandomIt first, RandomIt last, Comp&& comp = Comp(),
-        Proj&& proj = Proj())
+    HPX_DEPRECATED_V(
+        1, 8, "hpx::parallel::sort is deprecated, use hpx::sort instead")
+        typename util::detail::algorithm_result<ExPolicy, RandomIt>::type
+        sort(ExPolicy&& policy, RandomIt first, RandomIt last,
+            Comp&& comp = Comp(), Proj&& proj = Proj())
     {
         static_assert((hpx::traits::is_random_access_iterator<RandomIt>::value),
             "Requires a random access iterator.");
 
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
         return detail::sort<RandomIt>().call(std::forward<ExPolicy>(policy),
             first, last, std::forward<Comp>(comp), std::forward<Proj>(proj));
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic pop
+#endif
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+    ///////////////////////////////////////////////////////////////////////////
+    // DPO for hpx::sort
+    HPX_INLINE_CONSTEXPR_VARIABLE struct sort_t final
+      : hpx::functional::tag_fallback<sort_t>
+    {
+        // clang-format off
+        template <typename RandomIt,
+            typename Comp = hpx::parallel::v1::detail::less,
+            typename Proj = parallel::util::projection_identity,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<RandomIt>::value &&
+                parallel::traits::is_projected<Proj, RandomIt>::value &&
+                parallel::traits::is_indirect_callable<
+                    hpx::execution::sequenced_policy, Comp,
+                    parallel::traits::projected<Proj, RandomIt>,
+                    parallel::traits::projected<Proj, RandomIt>
+                >::value
+            )>
+        // clang-format on
+        friend RandomIt tag_fallback_dispatch(hpx::sort_t, RandomIt first,
```

Comment:
The algorithms in `namespace std` don't return anything (http://eel.is/c++draft/sort)

Reply:
Oh right my bad, I'll make the change. Thanks!